cmd/skopeo/list.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/urfave/cli"
+)
+
+type listOptions struct {
+	global      *globalOptions
+	image       *imageOptions
+	regUsername string
+	regPassword string
+}
+
+func listCmd(global *globalOptions) cli.Command {
+	sharedFlags, sharedOpts := sharedImageFlags()
+	imageFlags, imageOpts := imageFlags(global, sharedOpts, "", "")
+	opts := listOptions{
+		global: global,
+		image:  imageOpts,
+	}
+	return cli.Command{
+		Name:  "list",
+		Usage: "Get list of repos of REGISTRY-NAME",
+		Description: fmt.Sprint(`
+		Get list of repos of REGISTRY-NAME
+
+		Supports Docker v2 registries only.
```

Comment:
So should this subcommand be named `docker-list`? `registry-list`? `registry-catalog`?

Reply:
While this is for a Docker registry spec, it makes the most sense to stick with the vendor agnostic naming scheme.   Probably `registry-list` but `registry-catalog` would work equally well.

Reply:
- I’m not at all convinced this is even a bit easy to generalize:
  - The idea of a complete catalog per server clearly does not scale and it’s a wonder it’s been created at all, ever.
  - Already several transports in c/image don’t have repos/tags, or have them with very different syntax
  - Let alone the concept of a “registry” is something that just does not apply to local files or c/storage or BitTorrent-based transports
  So far c/image has not even tried to make sets of images / repositories / servers enumerable / generalizable beyond the docker/distribution protocol.
- Fine, supposing that this _could_ be generalizable: in that case the syntax of this command must be general and admit the various containers/image transports, or something like that; it can’t just take over the full set of strings for the first argument and interpret them as docker/distribution registries. So we would have `skopeo list --docker REGISTRY_NAME` or, maybe `skopeo list docker://REGISTRY_NAME` (but that runs into **huge difficulties** with consistency, because `docker://example.com` already has a meaning define to be equivalent to `docker://docker.io/library/example.com`, so that’s best avoided).